community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/helpers/RelationshipSelections.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.kernel.api.helpers;
+
+import org.neo4j.graphdb.Direction;
```

Comment:
We wanted to avoid depending on the graphdb API here, did we not?

Reply:
I'd change the design of this class to provide 6 static methods:

* `outgoingIterator(...)`
* `incomingIterator(...)`
* `allIterator(...)`
* `outgoingCursor(...)`
* `incomingCursor(...)`
* `allCursor(...)`

That way there would be no need to depend on `graphdb.Direction`, and if some caller has another encoding of direction it wouldn't need to translate.